library/Icingadb/Web/Controller.php

```diff
@@ -89,6 +97,39 @@ public function createPaginationControl(Paginatable $paginatable)
         $paginationControl->setDefaultPageSize($this->getPageSize(null));
         $paginationControl->setAttribute('id', $this->getRequest()->protectId('pagination-control'));
 
+        if (Url::fromRequest()->getParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM) === 'minimal') {
+            $hasLimitParam = Url::fromRequest()->hasParam(LimitControl::DEFAULT_LIMIT_PARAM);
+
+            if ($paginationControl->getDefaultPageSize() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() * 2);
+            }
+        }
+
+        $session = $this->Window()->getSessionNamespace(
+            $this->Window()->getId() . Url::fromRequest()->getPath()
```

Comment:
You should use `getContainerId()` here rather than just `getId()`. The latter is already baked into the session namespace identifier, but is also not of much interest because it's the container id which is the unique identifier we should rely on.

Oh and please add another hardcoded prefix into the mix, `3lkj2w3l4322l/icingaweb2/icingadb/hosts` is still too generic. What if someone else thinks of combining the container id with the url path? So please add a hardcoded name as well to make it less generic.

Reply:
✅